src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AddMissingInterpolationTokenTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines.CSharpAddMissingInterpolationToken,
+    Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines.CSharpAddMissingInterpolationTokeFixer>;
+
+namespace Microsoft.CodeQuality.Analyzers.UnitTests.QualityGuidelines
+{
+    public class AddMissingInterpolationTokenTests
+    {
+        [Fact]
+        public async Task HasValidVariableInScope_Diagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine([|""{x}""|]);
+    }
+}";
+
+            var fixedCode = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine($""{x}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
+        }
+
+        [Fact]
+        public async Task DoesNotHaveValidVariableInScope_NoDiagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine(""{y}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, code);
+        }
+
+        [Fact]
+        public async Task ContainsOnlyLiterals_NoDiagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        Console.WriteLine(""{0}"");
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(code, code);
+        }
+
+        [Fact]
+        public async Task ContainsLiteralAndBindableExpression_Diagnostic()
+        {
+            var code = @"
+using System;
+
+class Program
+{
+    public static void Main()
+    {
+        int x = 5;
+        Console.WriteLine([|""{x}, {0}""|]);
```

Comment:
Could we add a test for a case like: `Console.WriteLine(""{x}, {0}"", 1, 2)`? I would expect not to have the diagnostic but I could understand raising anyways because we are in a mixing scenario.

Reply:
Indeed the current implementation will raise a diagnostic. The suggestion by @sharwell at https://github.com/dotnet/roslyn-analyzers/issues/2767#issuecomment-656355022 was saying:

> My initial suggestion is exclude cases where **all** of the interpolation expressions are non-negative integer literals.

I'd say we